simulation/core/create_index.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Create a dataset index file from input parameters."""
+
+import json
+import os
+import random
+import string
+from typing import Optional
+
+from streaming.base.format import get_index_basename
+
+
+def get_random_foldername():
```

Comment:
Just tried using this and there's problems with using the `TemporaryDirectory()` because it falls out of scope once the function completes. Even returning the `TemporaryDirectory` from the function causes it to fall out of scope, meaning that it will get cleaned up before accessed in `SimulationDataset`. Because the index file creation functionality is separated from the instantiation of the StreamingDataset, I had to use actual folders instead of temp dirs.